<!-- -->
<launch>
  <arg name="role_name" default="ego_vehicle"/>
  <arg name="target_speed" default="20"/>
  <arg name="avoid_risk" default="True"/>
  <arg name="Kp_lateral" default="0.9"/>
  <arg name="Ki_lateral" default="0.0"/>
  <arg name="Kd_lateral" default="0.0"/>
  <arg name="Kp_longitudinal" default="0.206"/>
  <arg name="Ki_longitudinal" default="0.0206"/>
  <arg name="Kd_longitudinal" default="0.515"/>
  <arg name="control_time_step" default="0.05"/>

  <node pkg="carla_ad_agent" type="ad_agent.py" name="carla_ad_agent_$(arg role_name)" output="screen">
    <param name="target_speed" value="$(arg target_speed)" />
    <param name="role_name" value="$(arg role_name)" />
    <param name="avoid_risk" value="$(arg avoid_risk)" />
  </node>

  <node pkg="carla_ad_agent" type="local_planner.py" name="local_planner_$(arg role_name)" output="screen">
    <param name="role_name" value="$(arg role_name)" />
    <param name="Kp_lateral" value="$(arg Kp_lateral)" />
    <param name="Ki_lateral" value="$(arg Ki_lateral)" />
    <param name="Kd_lateral" value="$(arg Kd_lateral)" />
    <param name="Kp_longitudinal" value="$(arg Kp_longitudinal)" />
    <param name="Ki_longitudinal" value="$(arg Ki_longitudinal)" />
    <param name="Kd_longitudinal" value="$(arg Kd_longitudinal)" />
    <param name="control_time_step" value="$(arg control_time_step)" />
  </node>
</launch>

